### PR TITLE
Solve issue with Full Heal text bugfix

### LINKED ISSUE #818: Full Heal text isn't displayed correctly after applying "AI use of Full Heal does not cure confusion status" bugfix
This bugfix has a problem, and is that whenever the AI uses the Full Heal item the name is replaced with "?" instead:

[Example of Full Heal text being displayed incorrectly](https://user-images.githubusercontent.com/54337884/112940080-9de22780-90fa-11eb-9286-f703a3e7756c.png)

**NOTE:** The person who took the screenshot also tested the bugfix with a clean version of pokecrystal and this error still happened.

The cause of this problem is that, in `EnemyFullHeal`, register `a` contains the item used, gets stored in `[wCurEnemyItem]` and then `xor a` deletes this value.
```diff
 EnemyUsedFullHeal:
 	call AIUsedItemSound
 	call AI_HealStatus
 	ld a, FULL_HEAL
+	ld [wCurEnemyItem], a
+	xor a
+	ld [wEnemyConfuseCount], a
 	jp PrintText_UsedItemOn_AND_AIUpdateHUD
```
However, the first instruction in `PrintText_UsedItemOn_AND_AIUpdateHUD` tries store the value of `a` into `[wCurEnemyItem]` and, since it's 0, it can't print it properly.
```
; Parameter
; a = ITEM_CONSTANT
PrintText_UsedItemOn_AND_AIUpdateHUD:
	ld [wCurEnemyItem], a
	call PrintText_UsedItemOn
	jp AIUpdateHUD
```
**SOLUTION:** Only add `ld [wEnemyConfuseCount], a` before `ld a, FULL_HEAL` in `EnemyUsedFullHeal`. Register `a` already gets cleaned in `AI_HealStatus`, so we don't need to perform another `xor a`:
```diff
EnemyUsedFullHeal:
	call AIUsedItemSound
	call AI_HealStatus
+	ld [wEnemyConfuseCount], a
	ld a, FULL_HEAL
	jp PrintText_UsedItemOn_AND_AIUpdateHUD
```

### PR DESCRIPTION
**Pull request by Idain**
Fixes #818.

@Rangi42 and other mods, this PR shouldn't present merge conflicts, but if it does please let me know.

### CODE REVIEW

File: docs/bugs_and_glitches.md [hunk 1]
```diff
@@ -15,93 +15,95 @@ Fixes in the [multi-player battle engine](#multi-player-battle-engine) category
...
-- [Internal engine routines](#internal-engine-routines)
-  - [Saves corrupted by mid-save shutoff are not handled](#saves-corrupted-by-mid-save-shutoff-are-not-handled)
-  - [`ScriptCall` can overflow `wScriptStack` and crash](#scriptcall-can-overflow-wscriptstack-and-crash)
-  - [`LoadSpriteGFX` does not limit the capacity of `UsedSprites`](#loadspritegfx-does-not-limit-the-capacity-of-usedsprites)
-  - [`ChooseWildEncounter` doesn't really validate the wild Pokémon species](#choosewildencounter-doesnt-really-validate-the-wild-pokémon-species)
-  - [`TryObjectEvent` arbitrary code execution](#tryobjectevent-arbitrary-code-execution)
-  - [`ReadObjectEvents` overflows into `wObjectMasks`](#readobjectevents-overflows-into-wobjectmasks)
-  - [`ClearWRAM` only clears WRAM bank 1](#clearwram-only-clears-wram-bank-1)
-  - [`BattleAnimCmd_ClearObjs` only clears the first 6⅔ objects](#battleanimcmd_clearobjs-only-clears-the-first-6-objects)
+- [Bugs and Glitches](#bugs-and-glitches)
```
Comment (Rangi42) [inline docs/bugs_and_glitches.md:18]: Why are all of these lines modified?

Comment (Idain) [reply docs/bugs_and_glitches.md:18]: Oh my god, it must've been my Markdown plugin that did it. Sorry.

Comment (Idain) [reply docs/bugs_and_glitches.md:18]: Sorry, my Markdown plugin does it automatically and I didn't notice.

**Comment by Idain** [comment]
Ok, that should do it for now.

**Pull request merged**